### ISSUE TITLE
False negative used-before-assignment when the variable is assigned in try and used after except

### ISSUE THREAD

**Issue by alexNibbler** [pylint-dev/pylint#10477]
### Bug description

```python
import requests

def fun():
    try:
        response = requests.get("https://google.com", timeout=1)
    except Exception as exc:
        print(exc)
    return response # should raise E0601 used-before-assignment
```

### Configuration

```ini

```

### Command used

```shell
pylint script1.py
```

### Pylint output

```python
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

### Expected behavior

The given example is dummy, but it illustrates the case when a variable is set only in `try` block, `except` block doesn't raise exception and there is code using this variable after `try / except` block. 
I expect used-before-assignment to be caught here.

### Pylint version

```shell
pylint 3.3.7
python 3.13
```

### OS / Environment

MacOS 15.5

### Additional dependencies

```python

```

**Comment by mbyrnepr2** [comment]
I think you make a good point. variables defined in the `try` block should not be assumed to work later in the function.
The only thing I would quibble about is that it might be more accurate to define a new warning message for this since it isn't necessarily a false negative of `used-before-assignment` (since we don't know during static analysis if the `try` clause will succeed or not).

**Comment by alexNibbler** [comment]
Thanks for your reply, Mark! There is also message E0606 `possibly-used-before-assignment`, maybe that one fits better in this case.

**Comment by jacobtylerwalls** [comment]
Some history here. When I first implemented try/except fathoming in `used-before-assignment`, the decision we made was to just check the `except` blocks and let users treat `try` blocks as succeeding. Even this was a huge change for a minor release, so we stopped there.

This was discussed in a couple places, including here cc/ @DanielNoord:
- https://github.com/pylint-dev/pylint/pull/5384#issuecomment-978080181

I'm open to making this change, though, and v4 is probably a good time to get stricter. The implementation is probably not so difficult: we have helpers fathoming whether branches define a name, raise, or return in similar cases.

**Comment by jacobtylerwalls** [comment]
Duplicate of #2835
